engine-tests/src/tests/silo.rs

```diff
@@ -0,0 +1,752 @@
+    validate_address_balance_and_nonce(&runner, sender, INITIAL_BALANCE, INITIAL_NONCE.into());
+    validate_address_balance_and_nonce(&runner, receiver, ZERO_BALANCE, INITIAL_NONCE.into());
+}
+
+#[test]
+fn test_transfer_with_low_gas_limit() {
+    let (mut runner, mut signer, receiver) = initialize_transfer();
+    let sender = test_utils::address_from_secret_key(&signer.secret_key);
+    let caller: AccountId = CALLER_ACCOUNT_ID.parse().unwrap();
+
+    set_fixed_gas_cost(&mut runner, Some(FEE));
+    add_account_to_whitelist(&mut runner, caller);
+    add_address_to_whitelist(&mut runner, sender);
+
+    let transaction = |nonce| {
+        let mut tx = test_utils::transfer(receiver, TRANSFER_AMOUNT, nonce);
+        // it's not enough gas for common tx, but it doesn't matter if fixed cost is set
+        tx.gas_limit = 10_000.into();
+        tx
+    };
+
+    // validate pre-state
+    validate_address_balance_and_nonce(&runner, sender, INITIAL_BALANCE, INITIAL_NONCE.into());
+    validate_address_balance_and_nonce(&runner, receiver, ZERO_BALANCE, INITIAL_NONCE.into());
+
+    // make transfer
+    let result = runner.submit_with_signer(&mut signer, transaction).unwrap();
+    assert!(matches!(result.status, TransactionStatus::Succeed(_)));
+
+    validate_address_balance_and_nonce(
+        &runner,
+        sender,
+        INITIAL_BALANCE - FEE - TRANSFER_AMOUNT,
+        (INITIAL_NONCE + 1).into(),
+    );
+    validate_address_balance_and_nonce(&runner, receiver, TRANSFER_AMOUNT, INITIAL_NONCE.into());
+}
+
+#[test]
+fn test_relayer_balance_after_transfer() {
+    let (mut runner, mut source_account, receiver) = initialize_transfer();
+    let sender = test_utils::address_from_secret_key(&source_account.secret_key);
+    let caller: AccountId = CALLER_ACCOUNT_ID.parse().unwrap();
+    let relayer = sdk::types::near_account_to_evm_address(
+        runner.context.predecessor_account_id.as_ref().as_bytes(),
+    );
+    let transaction = |nonce| test_utils::transfer(receiver, TRANSFER_AMOUNT, nonce);
+
+    set_fixed_gas_cost(&mut runner, Some(FEE));
+    add_account_to_whitelist(&mut runner, caller);
+    add_address_to_whitelist(&mut runner, sender);
+
+    // validate pre-state
+    validate_address_balance_and_nonce(&runner, sender, INITIAL_BALANCE, INITIAL_NONCE.into());
+    validate_address_balance_and_nonce(&runner, receiver, ZERO_BALANCE, INITIAL_NONCE.into());
+
+    // do transfer
+    runner
+        .submit_with_signer(&mut source_account, transaction)
+        .unwrap();
+
+    // validate post-state
+    validate_address_balance_and_nonce(
+        &runner,
+        sender,
+        INITIAL_BALANCE - TRANSFER_AMOUNT - FEE,
+        (INITIAL_NONCE + 1).into(),
+    );
+    validate_address_balance_and_nonce(&runner, receiver, TRANSFER_AMOUNT, INITIAL_NONCE.into());
+    validate_address_balance_and_nonce(&runner, relayer, ZERO_BALANCE, INITIAL_NONCE.into());
```

Comment:
Why is the relayer balance still zero? The fee is simply burned?

Reply:
Good question. I think in terms of SILO we don't refund any gas to the relayer. But not sure. Good catch. Thanks. I will check it out.